tests/attestation_plugin_cert/enc/openssl_rsa.cpp

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/internal/raise.h>
+#include <openssl/bio.h>
+#include <openssl/bn.h>
+#include <openssl/evp.h>
+#include <openssl/pem.h>
+#include <openssl/rsa.h>
+#include <stdlib.h>
+#include <string.h>
+#include "rsa.h"
+
+oe_result_t generate_rsa_pair(
+    uint8_t** public_key,
+    size_t* public_key_size,
+    uint8_t** private_key,
+    size_t* private_key_size)
+{
+    oe_result_t result = OE_FAILURE;
+    uint8_t* local_public_key = nullptr;
+    uint8_t* local_private_key = nullptr;
+    int res = -1;
+    EVP_PKEY* pkey;
+    RSA* rsa;
+    BIO* bio = nullptr;
+    BIGNUM* e;
+    size_t local_public_key_size = 512;
+    size_t local_private_key_size = 2048;
+
+    // Generate RSA key
+    pkey = EVP_PKEY_new();
+    e = BN_new();
```

Comment:
need to test for null alloc or this test can crash.  Same in 38 etc.

Reply:
updated